cmd/generate-catalog/types.go

```diff
@@ -0,0 +1,262 @@
+package main
+
+import (
+	"fmt"
+	"slices"
+
+	semver "github.com/Masterminds/semver/v3"
+)
+
+type Input struct {
+	OldestSupportedVersion *semver.Version    `yaml:"oldest_supported_version"`
+	BrokenVersions         []*semver.Version  `yaml:"broken_versions"`
+	Images                 []InputBundleImage `yaml:"images"`
+}
+
+type InputBundleImage struct {
+	Image   string          `yaml:"image"`
+	Version *semver.Version `yaml:"version"`
+}
+
+type CatalogTemplate struct {
+	Schema  string         `yaml:"schema"`
+	Entries []CatalogEntry `yaml:"entries"`
+}
+
+type CatalogEntry any
+
+type Package struct {
+	Schema         string `yaml:"schema"`
+	Name           string `yaml:"name"`
+	DefaultChannel string `yaml:"defaultChannel"`
+	Icon           Icon   `yaml:"icon"`
+}
+
+type Icon struct {
+	Base64data string `yaml:"base64data"`
+	MediaType  string `yaml:"mediatype"`
+}
+
+type Channel struct {
+	Schema  string         `yaml:"schema"`
+	Name    string         `yaml:"name,omitempty"`
+	Package string         `yaml:"package,omitempty"`
+	Entries []ChannelEntry `yaml:"entries,omitempty"`
```

Comment:
I think, we need all those. Or is there a reason you added `omitempty`?
```suggestion
	Name    string         `yaml:"name"`
	Package string         `yaml:"package"`
	Entries []ChannelEntry `yaml:"entries"`
```